### PR TITLE
Add is defined to language reference.

### LINKED ISSUE #1419: Add 'is defined' to language reference

### PR DESCRIPTION
**Pull request by arnaudsjs**
Fixes #1419 #1454

- [x] add testcase for using is defined in implements (with implicit and explicit self)
- [x] add testcase for using is defined in if statement in an implementation refering to 
   - [x] block scope variable
   - [x] variable on implicit self
   - [x] variable on explicit self
   - [x] global in containing module

### CODE REVIEW

File: docs/language.rst [hunk 1]
```diff
@@ -171,8 +171,30 @@ Conditions can have the following forms
...
+
+    entity MonitoringServer:
+
+    end
+
+    Host.monitoring_server [0:1] -- MonitoringServer
+
+    implementation Host using monitoringConfig when monitoring_server is defined
+
+    implementation monitoringConfig:
```
Comment (bartv) [inline docs/language.rst]: This is invalid sytanx

File: docs/language.rst [hunk 2]
```diff
@@ -171,8 +171,30 @@ Conditions can have the following forms
...
+
+    end
+
+    entity MonitoringServer:
+
+    end
+
+    Host.monitoring_server [0:1] -- MonitoringServer
+
+    implementation Host using monitoringConfig when monitoring_server is defined
```
Comment (sanderr) [inline docs/language.rst]:
```suggestion
    implement Host using monitoringConfig when monitoring_server is defined
```

File: docs/language.rst [hunk 3]
```diff
@@ -171,8 +171,30 @@ Conditions can have the following forms
         | 'true'
         | 'false'
         | functioncall
+        | value 'is' 'defined'
         ;
 
+The ``is defined`` keyword checks if a value was assigned to an attribute or a relation of a certain entity. The following
```
Comment (sanderr) [inline docs/language.rst]: Not entirely sure, but I believe this is not limited to attributes and relations. The practical use cases might be limited, but I think you could use `is defined` on any value reference. I tried testing this using the if statement, but bumped into issue #1454.
@wouterdb can probably shed more light on this.

Comment (wouterdb) [reply docs/language.rst]: I fixed it to also use attributes.

**Pull request merged**